modules/nixos/fcitx5.nix

```diff
@@ -0,0 +1,29 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  inherit (config.catppuccin) sources;
+  cfg = config.i18n.inputMethod.fcitx5.catppuccin;
+  enable =
+    cfg.enable
+    && (
+      config.i18n.inputMethod.enabled == "fcitx5" # Deprecated but still available
+      || (config.i18n.inputMethod.enable && config.i18n.inputMethod.type == "fcitx5")
```

Comment:
we need to support 24.05 which doesn’t have `config.i18n.inputMethod.enable`. Evaluation of `config.i18n.inputMethod.enabled` should be avoided since it will be removed in the future. Do

```nix
cfg.enable
&& (
  config.i18n.inputMethod ? enable
  && config.i18n.inputMethod.enable
  && config.i18n.inputMethod.type == "fcitx5"
  || (!config.i18n.inputMethod ? enable && config.i18n.inputMethod.enabled == "fcitx5")
)
```